src/os.c

```diff
@@ -291,7 +291,16 @@ static void* mi_unix_mmap(size_t size, size_t try_alignment, int protect_flags)
       // TODO: if always failing due to permissions or no huge pages, try to avoid repeatedly trying?
       // Should we check this in _mi_os_init? (as on Windows)
       p = mi_unix_mmapx(size, try_alignment, protect_flags, lflags, fd);
+      #ifdef MADV_HUGEPAGE
+      if (p == MAP_FAILED) {
+        lflags = flags; // large page could be set to madvise only
+        p = mi_unix_mmapx(size, try_alignment, protect_flags, lflags, fd);
+        if (p == MAP_FAILED) return NULL; // we did what we could at this stage
+        else madvise(p, size, MADV_HUGEPAGE);
```

Comment:
The `MADV_HUGEPAGE` operation is available only if the kernel was configured with `CONFIG_TRANSPARENT_HUGEPAGE`, and I have no clue about the fallback path.

Reply:
As mimalloc deals with runtime environment configuration, it seems assumed the user needs to know its current settings kernel-wise. But to pass from madvise to always settings can be done via a basic file.